numba_dppy/dpnp_glue/dpnpimpl.py

```diff
@@ -0,0 +1,169 @@
+from numba.core.imputils import (lower_builtin)
+import numba_dppy.experimental_numpy_lowering_overload as dpnp_lowering
+from numba import types
+from numba.core.typing import signature
+from numba.core.extending import overload, register_jitable
+from . import stubs
+import numpy as np
+
+
+def get_dpnp_fptr(fn_name, type_names):
+    from . import dpnp_fptr_interface as dpnp_glue
+    f_ptr = dpnp_glue.get_dpnp_fn_ptr(fn_name, type_names)
+    return f_ptr
+
+@register_jitable
+def _check_finite_matrix(a):
+    for v in np.nditer(a):
+        if not np.isfinite(v.item()):
+            raise np.linalg.LinAlgError(
+                "Array must not contain infs or NaNs.")
+
+@register_jitable
+def _dummy_liveness_func(a):
+    """pass a list of variables to be preserved through dead code elimination"""
+    return a[0]
+
+
+
+class RetrieveDpnpFnPtr(types.ExternalFunctionPointer):
+    def __init__(self, fn_name, type_names, sig, get_pointer):
+        self.fn_name = fn_name
+        self.type_names = type_names
+        super(RetrieveDpnpFnPtr, self).__init__(sig, get_pointer)
+
+class _DPNP_EXTENSION:
+    def __init__(self, name):
+        dpnp_lowering.ensure_dpnp(name)
+
+    @classmethod
+    def dpctl_get_current_queue(cls):
+        ret_type  = types.voidptr
+        sig       = signature(ret_type)
+        return types.ExternalFunction("DPCTLQueueMgr_GetCurrentQueue", sig)
+
+    @classmethod
+    def dpctl_malloc_shared(cls):
+        ret_type  = types.voidptr
+        sig       = signature(ret_type, types.int64, types.voidptr)
+        return types.ExternalFunction("DPCTLmalloc_shared", sig)
+
+    @classmethod
+    def dpctl_queue_memcpy(cls):
+        ret_type  = types.void
+        sig       = signature(ret_type, types.voidptr, types.voidptr, types.voidptr, types.int64)
+        return types.ExternalFunction("DPCTLQueue_Memcpy", sig)
+
+    @classmethod
+    def dpctl_free_with_queue(cls):
+        ret_type  = types.void
+        sig       = signature(ret_type, types.voidptr, types.voidptr)
+        return types.ExternalFunction("DPCTLfree_with_queue", sig)
+
+
+    @classmethod
+    def dpnp_sum(cls, fn_name, type_names):
+        ret_type  = types.void
+        sig       = signature(ret_type, types.voidptr, types.voidptr, types.int64)
+        f_ptr     = get_dpnp_fptr(fn_name, type_names)
+        def get_pointer(obj):
+            return f_ptr
+        return types.ExternalFunctionPointer(sig, get_pointer=get_pointer)
+
+    @classmethod
+    def dpnp_eig(cls, fn_name, type_names):
```

Comment:
Since the overload is not yet ready, we should remove this function from this PR and bring it back in a follow up.